templates/common/on-prem/files/NetworkManager-resolv-prepender.yaml

```diff
@@ -20,13 +20,49 @@ contents:
     {{end -}}
     {{end -}}
 
+    function retry {
+        attempts=5
+        interval=1
+
+        local OPTIND
+        while getopts "a:i:" opt ; do
+        case "${opt}" in
+            a )
+                attempts="${OPTARG}"
+                ;;
+            i )
+                interval="${OPTARG}"
+                ;;
+            * )
+                ;;
+        esac
+        done
+        shift $((OPTIND-1))
+
+        rc=0
+        for attempt in $(seq "${attempts}")
+        do
+            >&2 echo "NM resolv-prepender: Attempt ${attempt}/${attempts} to execute \"$*\"..."
+
+            if "$@"; then
+                return 0
+            else
+                rc=$?
+                >&2 echo "NM resolv-prepender: Failed with exit code ${rc}, retrying \"$*\"..."
+                sleep "${interval}"
+            fi
+        done
+
+        return ${rc}
+    }
+
     function pull_baremetal_runtime_cfg_image {
         # This function must be executed as a background process and therefore the trap will only apply to this background
         # process. Do not let SIGTERM interrupt image pull due to image corruption issue [1]. Remove when issue is resolved.
         # [1] https://github.com/containers/podman/issues/14003
         trap "" SIGTERM
         >&2 echo "NM resolv-prepender: Starting download of baremetal runtime cfg image"
-        /usr/bin/podman pull --authfile /var/lib/kubelet/config.json {{ .Images.baremetalRuntimeCfgImage }}
+        retry -- /usr/bin/podman pull --authfile /var/lib/kubelet/config.json {{ .Images.baremetalRuntimeCfgImage }}
```

Comment:
how retrying baremetal image will help MCO to pull the osimage conatiner? what am I missing

Reply:
In the bugzilla reported you have the following error
```
Dec 28 12:56:33 ocpmaster02.cloudone.cloudz.local nm-dispatcher[2498]: Error: reading blob sha256:1e85b2b371520f19b4a9ae20e05de86b58a97910f5586fd03015f24fa23fd3bc: Get "https://cd[...]CSP66DGA": net/http: TLS handshake timeout
```
which comes from the following attempt
```
Dec 28 12:39:10 ocpmaster02.cloudone.cloudz.local nm-dispatcher[2498]: Trying to pull quay.io/openshift-release-dev/ocp-v4.0-art-dev@sha256:6ac6fca311193d6a987f2255769a8d7b93dc6435dccde51228dde19ab8ea45ad...
```

Image `sha256:6ac6fca311193d6a987f2255769a8d7b93dc6435dccde51228dde19ab8ea45ad` is an image of `baremetal-runtimecfg git ea6a9491`. Therefore this PR here is fixing a particular instance of what has been reported.

What am I missing?

Reply:
nothing. what im saying that it wont address the pivort failure. So lets open another issue, that talks about failure to fetch runitime-cfg, and link it to this PR?